mindocr/postprocess/det_fce_postprocess.py

```diff
@@ -295,66 +301,36 @@ def __init__(
-        # def __call__(self, preds, shape_list):
-        #     score_maps = preds['maps']
-        #     return self.get_boundary(score_maps, shape_list)
-
-    def __call__(self, preds, meta_info, **kwargs):
-        shape_list = meta_info[0].numpy()
+    def _postprocess(self, pred, **kwargs):
         score_maps = []
-        for value in preds:
+        for value in pred:
             value = value.asnumpy()
             cls_res = value[:, :4, :, :]
             reg_res = value[:, 4:, :, :]
             score_maps.append([cls_res, reg_res])
-        r = self.get_boundary(score_maps, shape_list)
+        r = self.get_boundary(score_maps, np.array([[0, 0, 1, 1]]))
         return {"polys": [row[0] for row in r], "scores": [row[1] for row in r]}
 
-    def resize_boundary(self, boundaries, scale_factor):
-        """Rescale boundaries via scale_factor.
-        Args:
-            boundaries (list[list[float]]): The boundary list. Each boundary
-            with size 2k+1 with k>=4.
-            scale_factor(ndarray): The scale factor of size (4,).
-        Returns:
-            boundaries (list[list[float]]): The scaled boundaries.
-        """
-        # assert check_argument.is_2dlist(boundaries)
-        # assert isinstance(scale_factor, np.ndarray)
-        # assert scale_factor.shape[0] == 4
-
-        boxes = []
-        scores = []
-        for b in boundaries:
-            sz = len(b)
-            valid_boundary(b, True)
-            scores.append(b[-1])
-            b = (
-                (np.array(b[: sz - 1]) * (np.tile(scale_factor[:2], int((sz - 1) / 2)).reshape(1, sz - 1)))
-                .flatten()
-                .tolist()
-            )
-            boxes.append(np.array(b).reshape([-1, 2]))
-
-        return np.array(boxes, dtype=np.float32), scores
-
     def get_boundary(self, score_maps, shape_list):
-        assert len(score_maps) == len(self.rescale_fields)
+        assert len(score_maps) == len(self.scales)
         # import pdb;pdb.set_trace()
         boundaries = []
         for idx, score_map in enumerate(score_maps):
-            scale = self.rescale_fields[idx]
+            scale = self.scales[idx]
             boundaries = boundaries + self._get_boundary_single(score_map, scale)
 
         # nms
         boundaries = poly_nms(boundaries, self.nms_thr)
 
-        # if rescale:
-        # import pdb;pdb.set_trace()
-        boundaries, scores = self.resize_boundary(boundaries, (1 / shape_list[0, 2:]).tolist()[::-1])
-
-        # boxes_batch = [dict(points=boundaries, scores=scores)]
-        boxes_batch = [[boundaries, scores]]
+        boxes = []
+        scores = []
+        for b in boundaries:
+            sz = len(b)
+            valid_boundary(b, True)
+            scores.append(b[-1])
+            boundary = np.array(b[: sz - 1]).flatten().tolist()
+            boxes.append(np.array(boundary).reshape([-1, 2]))
+        boxes_batch = [[np.array(boxes, dtype=np.float32), scores]]
         return boxes_batch
```

Comment:
you pack data here and then unpack it in `_postprocess`: `{"polys": [row[0] for row in r], "scores": [row[1] for row in r]}`. I think it will be easier to return both variables and skip pack-unpack stage.

Reply:
done